### DISCUSSION TITLE
Translation of Songs

### DISCUSSION THREAD

**Discussion by FelixHeppner** [ChurchApps/ChurchAppsSupport#320] [FreeShow] [answered]
Is there or will there be a feature to have song translations in  free show? 
I can write both languages on each slide and style the text accordingly but managing and changing these styles might become difficult. 
I am not aware how ProPresenter implements this. We are currently using Quelea Projection (an open source tool built with java fx).
We definitely need the possibility to show translations of Songs on the main screen. And if possible we do not want translations on the stage screen.  I am still discovering a lot of cool features of FreeShow so I might have missed this feature or how it works here so far?

**Comment by vassbo** [comment]
Please check out this: https://github.com/ChurchApps/FreeShow/issues/375
And let me know if it helps or not.

**Comment by FelixHeppner** [comment]
Wow. This is kind of not obvious at first but it will work. Thanks.  And the styling using template will allow for pretty clear maintenance.  

Can I somehow switch between  a stage layout/style/template with the main output and (hopefully) the next slide displayed, even if next slide is an image, and a layout/style/template showing text only for lyrics and especially only the text without translation? I can assign a style to an output with an action on first slide, but I did not mange to somehow switch the visibility in a way that works for me. 

The interaction an "correct" application for templates, styles and layouts in regards to stage display is not yet clear to me.

**Reply by vassbo** [reply]
Great! There are a few ways to do that, but I guess the easiest would be to create two different stage layouts in the "Stage" view, and use the "Mirror" item inside two templates, set to the different stage layouts.
Or you can try creating a template with just the mirror item to show the next slide text, and change the output layout.
You will have to experiment a bit yourself to get it exactly how you want it.

**Reply by FelixHeppner** [reply, marked as answer]
Okay. I will make some more experiments...

I did not manage to change switch or somehow affect the layout shown on the output so far. I am not even aware how to apply template to the stage output layout. I can create multiple layout but how are these activated (on hdmi out)? I can create styles and let these override/apply a template and apply the style to the stage output in an action. But no effect so far.

Thanks for you answers and the awesome software. I will be afk a few days. If I can figure this out I will write something to be included in the docs hopefully :-)
